protocols/luci-proto-wireguard/luasrc/model/cbi/admin_network/proto_wireguard.lua

```diff
@@ -81,10 +81,22 @@ preshared_key = section:taboption(
             "cryptography for post-quantum resistance.")
 )
 preshared_key.password = true
-preshared_key.datatype = "and(base64,rangelength(44, 44))"
+preshared_key.datatype = "and(base64,rangelength(44,44))"
 preshared_key.optional = true
 
 
+fwmark = section:taboption(
+  "advanced",
+  Value,
+  "fwmark",
+  translate("Firewall Mark"),
+  translate("Optional. 32-bit hexadecimal mark for outgoing encrypted " ..
+            "packets. Enter value including leading <code>0x</code>.")
+)
+fwmark.datatype = "and(hexstring,rangelength(0,8))"
```

Comment:
Not exactly, but very close. Here is the version that worked.

````
local datatypes  = require("luci.cbi.datatypes")
datatypes["fwmark"] = function(val)
	if val then
		if (val:match("^0x[a-fA-F0-9]+$") ~= nil) and (#val < 11) then
			return true
		end
		return datatypes["range"](val, 0, 4294967296)
	end
end
````

@jow- and/or @hnyman How is you stance regarding "local datatypes" versus global datatypes? Would `fwmark` be handy in other places, too? If not, is a local datatype acceptable?